osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/utils/csv/CsvWriter.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.shared.utils.csv;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.Assert;
+
+public class CsvWriter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CsvWriter.class);
+
+    private static final String COMMA = ",";
+    private static final String CSV = ".csv";
+    private static final String DATE_TIME_FORMAT = "yyyyMMdd-HHmmss";
+
+    /**
+     * Create or validate the folder where a CSV file is going to be stored.
+     *
+     * @param csvFileStorageLocation
+     *            The path to the folder.
+     */
+    public void checkCsvFileStorageLocation(final String csvFileStorageLocation) {
+        new File(csvFileStorageLocation).mkdirs();
```

Comment:
Will this throw an exception when creating the folders fails? Or will the mkdirs() call just return false?

Reply:
Only security-related issues throw exceptions.
```
mkdirs
public boolean mkdirs()
Creates the directory named by this abstract pathname, including any necessary but nonexistent parent directories. Note that if this operation fails it may have succeeded in creating some of the necessary parent directories.
Returns:
true if and only if the directory was created, along with all necessary parent directories; false otherwise
Throws:
SecurityException - If a security manager exists and its SecurityManager.checkRead(java.lang.String) method does not permit verification of the existence of the named directory and all necessary parent directories; or if the SecurityManager.checkWrite(java.lang.String) method does not permit the named directory and all necessary parent directories to be created
```